### PR TITLE
Fix periodic re-reconciles of NNCPs with name equal to node

### PR DESCRIPTION
**Pull request by rhrazdil**
Policies that are named the same as worker nodes are getting
periodically re-reconciled.
It seems that the `For(&corev1.Node{})` creates a watch of it's own
that overrides the builder.WithPredicates(onLabelsUpdatedForThisNode)
defined in the subsequent `Watches(...)` statement.

This commit creates the controller manually without using
controller-runtime builder.

Signed-off-by: Radim Hrazdil <rhrazdil@redhat.com>

**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind enhancement

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**Release note**:

```release-note
Fix issue with NNCP named the same as nodes geting periodically re-reconciled
```

### CODE REVIEW

File: controllers/handler/nodenetworkconfigurationpolicy_controller.go [hunk 1]
```diff
@@ -258,13 +258,19 @@ func (r *NodeNetworkConfigurationPolicyReconciler) SetupWithManager(mgr ctrl.Man
...
-		Complete(r)
+	c, err := controller.New("node_labels", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return errors.Wrap(err, "failed to create node_labels controller to NNCP Reconciler watching node label changes")
+	}
+	err = c.Watch(
+		&source.Kind{Type: &corev1.Node{}},
+		handler.EnqueueRequestsFromMapFunc(allPolicies),
+		onLabelsUpdatedForThisNode,
+	)
```
Comment (qinqon) [inline controllers/handler/nodenetworkconfigurationpolicy_controller.go:269]: We don't need `Complete(r)` ?

Comment (rhrazdil) [reply controllers/handler/nodenetworkconfigurationpolicy_controller.go:269]: We're creating the controller with `controller.New` on line, so we don't need Complete, as complete is used to instantiate the controller from the controller builder

**Pull request merged**